app/Main.hs

```diff
@@ -1,13 +1,25 @@
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
 module Main where
 
 import Web.Spock
 import Web.Spock.Config
+import Control.Monad.IO.Class (liftIO)
+import Data.Yaml.Config (loadYamlSettings, useEnv)
+import Data.Aeson (parseJSON, FromJSON, withObject, (.:))
+
+data AppSettings = AppSettings { appPort :: Int }
+
+instance FromJSON AppSettings where
+  parseJSON = withObject "AppSettings" $ \o -> do
```

Comment:
You can't define type signatures inside of instances without a language extension (which just removes them) the type of parseJSON is already defined in the class `FromJSON`: https://hackage.haskell.org/package/aeson-1.2.1.0/docs/Data-Aeson.html#t:FromJSON